[ADP-3479] Add input bip32 list to Transaction type

paolino opened the pull request:
- Add bip32 input paths to unsigned json to easy external signing

ADP-3479

paolino committed:
Add input bip32 list to Transaction type

abailly reviewed (commented):
I think we need a roundtrip property test for BIP32Path <-> JSON encoding, and I would add the `decodeBip32/encodeBIP32` as proper `FromJSON/ToJSON` instances to the `BIP32Path`. With `ToJSON/FromJSON/Arbitrary` instances available, we can leverage `hspec-golden-aeson` to add standard tests.

abailly committed:
Add roundtrip JSON property tests for BIP32Path

The ToJSON/FromJSON/Arbitrary instances for BIP32Path should probably
live alongside the corresponding datatype definition but it's good
enough for now. Interestingly, writing the property showed a bug: The
parsing of `Root` path was incorrect!

Anviking reviewed (approved):
:+1:
